scenarios/Makefile

```diff
@@ -2,6 +2,20 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 
+OSTYPE := $(shell uname -s)
+ifeq ($(OSTYPE),Linux)
+NPROCS := $(shell nproc)
+else ifeq ($(OSTYPE),Darwin) # Assume Mac OS X
+NPROCS := $(shell sysctl -n hw.physicalcpu)
+else
+NPROCS := 1
+endif
+
+# Extracts the services from docker-compose, this is needed to avoid pulling
+# sdfabric-onos images built locally, when downloading dependencies for docker-compose
+STC_COMPOSE_SERVICES = $(shell python3 -c "import yaml; f = open('docker-compose.yml'); y = yaml.safe_load(f); print(' '.join(y['services'])); f.close();")
```

Comment:
Does this require pyyaml to be pre-installed? It would be good not too have additional dependencies, but if we need to, it should be mentioned in the README.

Reply:
Yes, let me try to come up with a version using `awk`. I would like to avoid putting dependencies, otherwise, I'll create a script that creates a virtualenv.